frontend/src/metabase/visualizations/components/settings/ChartSettingNestedSettings.jsx

```diff
@@ -55,15 +55,23 @@ const chartSettingNestedSettings =
         }
       };
 
-      handleChangeSettingsForObjectKey = (objectKey, changedSettings) => {
-        const { onChange } = this.props;
-        const objectsSettings = this.props.value || {};
-        const objectSettings = objectsSettings[objectKey] || {};
-        const newSettings = updateSettings(objectSettings, changedSettings);
-        onChange({
-          ...objectsSettings,
-          [objectKey]: newSettings,
-        });
+      handleChangeSettingsForObjectKey = (changedKey, changedSettings) => {
+        const { objects, onChange } = this.props;
+        const oldSettings = this.props.value || {};
+        const newSettings = objects.reduce((newSettings, object) => {
+          const currentKey = getObjectKey(object);
+          const objectSettings = getObjectSettings(oldSettings, object);
+          if (currentKey === changedKey) {
```

Comment:
An explanatory comment with scenario in which this condition can be false would be very useful here :pray: